flow/projecttracking/jira/jira.py

```diff
@@ -0,0 +1,376 @@
+                story_detail.get('url', '')), 'ERROR')
+            commons.print_msg(Jira.clazz, method, e, 'ERROR')
+            exit(1)
+        
+        if resp.status_code == 200:
+            json_data = json.loads(resp.text)
+            commons.print_msg(Jira.clazz, method, json_data)
+        else:
+            commons.print_msg(Jira.clazz, method, "Failed retrieving story detail from call to {url}. \r\n "
+                                                         "Response: {response}".format(url=story_detail.get('url', ''),
+                                                                                       response=resp.text), 'WARN')
+
+        commons.print_msg(Jira.clazz, method, 'end')
+        return json_data
+
+    def tag_stories_in_commit(self, story_list):
+        method = 'tag_stories_in_commit'
+        commons.print_msg(Jira.clazz, method, 'begin')
+
+        for story in story_list:
+            version = '{0}-{1}'.format(self.config.project_name, self.config.version_number)
+            self._add_version_to_project(version)
+            self._add_version_to_story(story, version)
+
+        commons.print_msg(Jira.clazz, method, 'end')
+
+    def _add_version_to_project(self, version):
+        method = '_add_version_to_project'
+        commons.print_msg(Jira.clazz, method, 'begin')
+
+        for idx, project_id in enumerate(self.project_keys):
+            version_to_post = Object()
+            version_to_post.projectId = project_id[0]
+            version_to_post.name = version.lower()
+
+            jira_url = "{url}/rest/api/3/version".format(url=Jira.jira_url)
+
+            headers = {'Content-type': 'application/json', 'Accept': 'application/json',
+                       'Authorization': 'Basic {0}'.format(Jira.jira_basic_auth)}
+
+            commons.print_msg(Jira.clazz, method, version_to_post.to_JSON())
+
+            try:
+                resp = requests.post(jira_url, version_to_post.to_JSON(), headers=headers, timeout=self.http_timeout)
+
+                if resp.status_code != 201:
+                    commons.print_msg(Jira.clazz, method, "Unable to create version {version} for project {project} \r\n "
+                                                          "Response: {response}".format(version=version, project=project_id[1], response=resp.text), 'WARN')
+                else: 
+                    commons.print_msg(Jira.clazz, method, resp.text)
+            except requests.ConnectionError as e:
+                commons.print_msg(Jira.clazz, method, 'Connection error. ' + str(e), 'WARN')
+            except Exception as e:
+                commons.print_msg(Jira.clazz, method, "Unable to create version {version} for project {project}".format(
+                    version=version, project=project_id[1]), 'WARN')
+                commons.print_msg(Jira.clazz, method, e, 'WARN')
+
+        commons.print_msg(Jira.clazz, method, 'end')
+
+    def _add_version_to_story(self, story_id, version):
```

Comment:
Seeing an issue when trying to label stories with the version.  It successfully creates the version,

![image](https://user-images.githubusercontent.com/6393612/117985797-0251f280-b307-11eb-9344-a86dbd3c9608.png)



```shell
[DEBUG] Jira         extract_story_id_from_commit_messages begin
[DEBUG] Jira         extract_story_id_from_commit_messages Story list: ['JIRA-1035', 'JIRA-1034']
[DEBUG] Jira         extract_story_id_from_commit_messages end
[DEBUG] Jira         tag_stories_in_commit               begin
[DEBUG] Jira         _add_version_to_project             begin
[DEBUG] Jira         _add_version_to_project             {
    "name": "ci-flow-test-java-jira-v1.1.0",
    "projectId": "10031"
}
[DEBUG] Jira         _add_version_to_project             {"self":"https://redacted.atlassian.net/rest/api/3/version/10004","id":"10004","name":"ci-flow-test-java-jira-v1.1.0","archived":false,"released":false,"projectId":10031}
[DEBUG] Jira         _add_version_to_project             end
[DEBUG] Jira         _add_version_to_story               begin
[DEBUG] Jira         _add_version_to_story               https://redacted.atlassian.net/rest/api/3/issue/JIRA-1035
[WARN]  Jira         _add_version_to_story               Unable to add version CI-Flow-Test-Java-Jira-v1.1.0 to issue JIRA-1035
 Response: {"errorMessages":["Unexpected character ('u' (code 117)): expected a valid value (number, String, array, object, 'true', 'false' or 'null')\n at [Source: org.apache.catalina.connector.CoyoteInputStream@48a60219; line: 1, column: 2]"]}
[DEBUG] Jira         _add_version_to_story               end
[DEBUG] Jira         _add_version_to_project             begin
[DEBUG] Jira         _add_version_to_project             {
    "name": "ci-flow-test-java-jira-v1.1.0",
    "projectId": "10031"
}
[WARN]  Jira         _add_version_to_project             Unable to create version CI-Flow-Test-Java-Jira-v1.1.0 for project JIRA
 Response: {"errorMessages":[],"errors":{"name":"A version with this name already exists in this project."}}
[DEBUG] Jira         _add_version_to_project             end
[DEBUG] Jira         _add_version_to_story               begin
[DEBUG] Jira         _add_version_to_story               https://redacted.atlassian.net/rest/api/3/issue/JIRA-1034
[WARN]  Jira         _add_version_to_story               Unable to add version CI-Flow-Test-Java-Jira-v1.1.0 to issue JIRA-1034
 Response: {"errorMessages":["Unexpected character ('u' (code 117)): expected a valid value (number, String, array, object, 'true', 'false' or 'null')\n at [Source: org.apache.catalina.connector.CoyoteInputStream@6249b5a0; line: 1, column: 2]"]}
[DEBUG] Jira         _add_version_to_story               end
[DEBUG] Jira         tag_stories_in_commit               end
```

Reply:
Fixed this issue too, needed to perform a json.dumps on the data object before sending it to the api